hudi-common/src/test/java/org/apache/hudi/common/config/TestConfigProperty.java

```diff
@@ -171,4 +171,28 @@ public void testAdvancedValue() {
     assertTrue(FAKE_BOOLEAN_CONFIG.markAdvanced().isAdvanced());
     assertTrue(FAKE_BOOLEAN_CONFIG_NO_DEFAULT.markAdvanced().isAdvanced());
   }
+
+  @EnumDescription("Test enum description.")
+  public enum TestEnum {
```

Comment:
Could you add tests for case-insensitive configs?  e.g., setting `test_val_a` should use `TEST_VAL_A`.  Or if `TEST_VAL_A("val_a")` is defined, `val_a` should transform to `TEST_VAL_A`.

Reply:
That has to happen in the getter and/or the setter, it won't happen in configproperty